### PR TITLE
fix(sdk): add special Exceptions for the manager logic, when trying to connect to a gone service

### PR DESCRIPTION
**Pull request by kptkin**
Fixes WB-12268

Description
-----------
What does the PR do?

- Adds more fine grained errors to the manager to help debug cases where the token is stale (since the service closed) vs real manager issues
- capture error context more cleanly and verbose 
- add validators to time-based settings

Testing
-------
How was this PR tested?

Checklist
-------
- [ ] Include reference to internal ticket "Fixes WB-NNNN" and/or GitHub issue "Fixes #NNNN" (if applicable)
- [ ] Ensure PR title compliance with the [conventional commits standards](https://github.com/wandb/wandb/blob/main/CONTRIBUTING.md#conventional-commits)

### CODE REVIEW

File: wandb/sdk/service/service.py [hunk 1]
```diff
@@ -26,6 +22,24 @@
     from wandb.sdk.wandb_settings import Settings
 
 
+class ServiceStartProcessError(Exception):
```
Comment (kptkin) [inline wandb/sdk/service/service.py:25]:
```suggestion
class ServiceStartProcessError(Error):
```

File: wandb/sdk/service/service.py [hunk 2]
```diff
@@ -26,6 +22,24 @@
...
+    """Raised when a known error occurs when launching wandb service"""
+
+    pass
+
+
+class ServiceStartTimeoutError(Exception):
```
Comment (kptkin) [inline wandb/sdk/service/service.py:31]:
```suggestion
class ServiceStartTimeoutError(Error):
```

File: wandb/sdk/service/service.py [hunk 3]
```diff
@@ -26,6 +22,24 @@
...
+    """Raised when service start times out"""
+
+    pass
+
+
+class ServiceStartPortError(Exception):
```
Comment (kptkin) [inline wandb/sdk/service/service.py:37]:
```suggestion
class ServiceStartPortError(Error):
```

File: wandb/sdk/service/service.py [hunk 4]
```diff
@@ -12,12 +12,8 @@
...
 from typing import TYPE_CHECKING, Any, Dict, Optional
 
-from ...errors import (
-    ServiceStartPortError,
-    ServiceStartProcessError,
-    ServiceStartTimeoutError,
-)
-from ...util import sentry_reraise, sentry_set_scope
+from wandb.util import sentry_reraise, sentry_set_scope
+
```
Comment (kptkin) [inline wandb/sdk/service/service.py:16]: from wandb.errors import Error

File: wandb/sdk/service/service.py [hunk 5]
```diff
@@ -12,12 +12,8 @@
 import time
 from typing import TYPE_CHECKING, Any, Dict, Optional
 
-from ...errors import (
-    ServiceStartPortError,
-    ServiceStartProcessError,
-    ServiceStartTimeoutError,
-)
-from ...util import sentry_reraise, sentry_set_scope
+from wandb.util import sentry_reraise, sentry_set_scope
```
Comment (kptkin) [inline wandb/sdk/service/service.py:15]:
```suggestion
from wandb.errors import Error
from wandb.util import sentry_reraise, sentry_set_scope
```

File: wandb/sdk/wandb_manager.py [hunk 6]
```diff
@@ -7,18 +7,33 @@
...
 from typing import TYPE_CHECKING, Any, Callable, Dict, Optional
 
+import psutil
+
 import wandb
 from wandb import env, trigger
 from wandb.sdk.lib.exit_hooks import ExitHooks
 from wandb.sdk.lib.import_hooks import unregister_all_post_import_hooks
 from wandb.sdk.lib.proto_util import settings_dict_from_pbmap
+from wandb.util import sentry_reraise
```
Comment (kptkin) [inline wandb/sdk/wandb_manager.py:17]:
```suggestion
from wandb.errors import Error
from wandb.util import sentry_reraise
```

File: wandb/sdk/wandb_manager.py [hunk 7]
```diff
@@ -7,18 +7,33 @@
...
 
 if TYPE_CHECKING:
     from wandb.sdk.service import service
     from wandb.sdk.service.service_base import ServiceInterface
     from wandb.sdk.wandb_settings import Settings
 
 
+class ManagerConnectionError(Exception):
```
Comment (kptkin) [inline wandb/sdk/wandb_manager.py:25]:
```suggestion
class ManagerConnectionError(Error):
```

File: wandb/sdk/wandb_manager.py [hunk 8]
```diff
@@ -93,8 +109,27 @@ class _Manager:
...
+        svc_iface = self._get_service_interface()
+
+        try:
+            svc_iface._svc_connect(port=port)
+        except ConnectionRefusedError as e:
+            if not psutil.pid_exists(self._token.pid):
+                message = (
+                    "Connection to wandb service failed "
+                    "since the process is not available. "
+                    # "See [TODO] for more details"
```
Comment (kptkin) [inline wandb/sdk/wandb_manager.py:123]:
```suggestion
```

**Review by dmitryduev — APPROVED** [review]
LGTM ;)

**Pull request merged**